lib/propshaft/compilers/css_asset_urls.rb

```diff
@@ -1,13 +1,40 @@
+# frozen_string_literal: true
+
 class Propshaft::Compilers::CssAssetUrls
   attr_reader :assembly
 
   def initialize(assembly)
     @assembly = assembly
   end
 
-  def compile(input)
-    input.gsub(/asset-path\(["']([^"')]+)["']\)/) do |match|
-      %[url("#{assembly.config.prefix}/#{assembly.load_path.find($1).digested_path}")]
+  def compile(logical_path, input)
+    input.gsub(/url\(\s*["']?([^"'\s)]+)\s*["']?\)/.freeze) do
+      match = Regexp.last_match
+
+      if match[1] =~ /^(http|data|#)/
+        match[0]
+      else
+        resolved_path = resolve_path(logical_path.dirname, match[1])
+        resolved_path, mask = extract_svg_mask(resolved_path.to_s)
+        %[url("#{assembly.config.prefix}/#{assembly.load_path.find(resolved_path).digested_path}#{mask}")]
```

Comment:
What do we want to happen if we can’t find the `resolved_path`? I feel like it should blow up, your CSS shouldn’t compile.

I think it _will_ blow up as written — `find` returns nil and you’ll get a `NoMethodError` for `digested_path`. But that might confuse people and make them think Propshaft is broken. So seems to me that we need either to raise in `find` or define a `find!` method that raises (if there are legit non-raising cases).